src/router/definitions.ts

```diff
@@ -0,0 +1,54 @@
+export type Route =
+  | { type: "404" }
+  | {
+      type: "faucet";
+      params: {
+        activeView:
+          | { type: "form" }
+          | { type: "withdraw"; params: { amount: string | null } };
+      };
+    }
+  | { type: "home" }
+  | { type: "boot" }
+  | { type: "profile"; params: { addressOrName: string } }
+  | { type: "projects"; params: ProjectsParams }
+  | {
+      type: "registration";
+      params: {
+        activeView:
+          | {
+              type: "validateName";
+            }
+          | {
+              type: "checkNameAvailability";
+              params: { nameOrDomain: string; owner: string | null };
+            }
+          | {
+              type: "register";
+              params: { nameOrDomain: string; owner: string | null };
+            }
+          | {
+              type: "view";
+              params: { nameOrDomain: string; retry: boolean };
+            };
+      };
+    }
+  | { type: "seeds"; params: { host: string } }
+  | { type: "vesting" };
```

Comment:
I'd rename `type` to `resource`, or `name`, since these routes are all of the same type.
I'm also wondering why `activeView` is not called `view`? Is there an `inactiveView` as well?

Reply:
I think we started using `type` in Upstream, because the route itself is a type union. But yeah, the attribute name also could be something else. My first choice would still be `type`, second `resource`. As for `activeView`, maybe more appropriate would be `subView` or `subRoute`.

Reply:
fixed by 39a604e75653067708bf34b5b1c51f0b2abd8d8d, 72690b7f2496b248c236bfcf7c42cb611942a423 and finally 76c9146346648573fec1ee5584157c85485b9429